nvflare/fuel/f3/cellnet/new_cell.py

```diff
@@ -0,0 +1,180 @@
+        request.set_header(MessageHeaderKey.CHANNEL, channel)
+        topic = request.get_header(StreamHeaderKey.TOPIC)
+        request.set_header(MessageHeaderKey.TOPIC, topic)
+        req_id = request.get_header(MessageHeaderKey.REQ_ID)
+        response = self.cb(request)
+        response.add_headers(
+            {
+                MessageHeaderKey.CHANNEL: channel,
+                MessageHeaderKey.TOPIC: topic,
+                MessageHeaderKey.FROM_CELL: self.my_info.fqcn,
+                MessageHeaderKey.ORIGIN: self.my_info.fqcn,
+                MessageHeaderKey.DESTINATION: origin,
+                MessageHeaderKey.TO_CELL: origin,
+                MessageHeaderKey.REQ_ID: req_id,
+                MessageHeaderKey.MSG_TYPE: MessageType.REPLY,
+                MessageHeaderKey.ROUTE: [(self.my_info.fqcn, time.time())],
+                StreamMessageKey.REQ_ID: stream_req_id,
+            }
+        )
+        messagesend_future = self.nice_cell.send_blob(channel, topic, origin, response)
+
+
+class NewCell(StreamCell):
+    def __init__(self, *args, **kwargs):
+        self.core_cell = Cell(*args, **kwargs)
+        super().__init__(self.core_cell)
+        self.req_reg = Registry()
+        self.requests_dict = dict()
+        self.logger = logging.getLogger(self.__class__.__name__)
+
+    def start(self):
+        self.core_cell.start()
+
+    def stop(self):
+        self.core_cell.stop()
+
+    def is_cell_connected(self, target_fqcn):
+        return self.core_cell.is_cell_connected(target_fqcn)
+
+    def broadcast_multi_requests(self, *args, **kwargs):
+        return self.core_cell.broadcast_multi_requests(*args, **kwargs)
+
+    def get_root_url_for_child(self):
+        return self.core_cell.get_root_url_for_child()
+
+    def fire_and_forget(
+        self, channel: str, topic: str, targets: Union[str, List[str]], message: Message, optional=False
+    ) -> Dict[str, str]:
+        return self.core_cell.fire_and_forget(
+            channel=channel, topic=topic, targets=targets, message=message, optional=optional
+        )
+
+    def get_fqcn(self):
+        return self.core_cell.get_fqcn()
+
+    def get_internal_listener_url(self) -> Union[None, str]:
+        """Get the cell's internal listener url.
+
+        This method should only be used for cells that need to have child cells.
+        The url returned is to be passed to child of this cell to create connection
+
+        Returns: url for child cells to connect
+
+        """
+        return self.core_cell.get_internal_listener_url()
+
+    def send_request(self, channel, target, topic, request, timeout=None, optional=False):
+        if channel != CellChannel.SERVER_COMMAND:
+            return self.core_cell.send_request(
+                channel=channel, target=target, topic=topic, request=request, timeout=timeout, optional=optional
+            )
+
+        self.register_blob_cb("*", "*", self._process_reply)  # this should be one-time registration
```

Comment:
This defeats purpose of the registry in StreamCell. Should be able to use one registry.